replica/database.cc

```diff
@@ -2591,6 +2591,16 @@ dht::token_range_vector database::get_keyspace_local_ranges(sstring ks) {
     return find_keyspace(ks).get_vnode_effective_replication_map()->get_ranges(my_address);
 }
 
+std::optional<dht::token_range_vector> database::maybe_get_keyspace_local_ranges(sstring ks) {
+    const auto& keyspace = find_keyspace(ks);
+    if (keyspace.get_replication_strategy().is_per_table()) {
+        // return nullopt if each tables have their own effective_replication_map
+        return std::nullopt;
+    }
+    auto my_address = get_token_metadata().get_topology().my_address();
+    return find_keyspace(ks).get_vnode_effective_replication_map()->get_ranges(my_address);
+}
```

Comment:
nit, you already found keyspace above, you don't have to do it again.

Reply:
Ack. Fixed it with a new version.